x-pack/plugins/alerting/server/lib/last_run_status.ts

```diff
@@ -42,9 +51,22 @@ export const lastRunFromState = (stateWithMetrics: RuleTaskStateAndMetrics): ILa
     outcomeMsg = translations.taskRunner.warning.maxExecutableActions;
   }
 
+  // Overwrite outcome to be error if last run reported any errors
+  if (errors.length > 0) {
+    outcome = RuleLastRunOutcomeValues[2];
+  }
+
+  // Optionally merge outcome messages reported by
+  // rule execution to the Framework's outcome message
+  if (outcomeMsg.length > 0 && outcomeMessage.length > 0) {
+    outcomeMsg = `${outcomeMsg} - ${outcomeMessage}`;
+  } else if (outcomeMessage.length > 0) {
+    outcomeMsg = outcomeMessage;
+  }
```

Comment:
My impression from the meeting was that the proposed behavior is "If outcome message is set by Security via the last run service, don't rewrite it", so it should be something like that in the code:

```
const finalOutcomeMessage = executorOutcomeMessage || frameworkOutcomeMessage || '';
```

Here the code shows that Framework sets the outcome message in 2 certain cases, so it could justify the concatenation. However, it would change the existing behavior in Security Solution.

I don't have a strong opinion on this and would appreciate any other thoughts.

Reply:
I think the concatenation is not a problem (our user will get more info), but if we do that I will prefer to convert the outcomeMsg to string[]. It will be nicer for the UI to re-render. What do you think?

Reply:
it won't be a problem in our hand yet since we did not update our UI to show this info, but for you it might?

Reply:
we can create an issue on the API to always give it back as Array

Reply:
No objections to making it an array in the rule, we can concatenate it back to a single string on read before returning from the detections API. We need it to be a string there.

In the last run service, we don't need the `outcomeMessage` to be an array though.

Reply:
> our user will get more info

++ that's a good argument 👍

Reply:
@XavierM Set the outcomeMsg in the rule to be `string[]`, but it is still a `string` in the `RuleExecutionService` since we expect the last run execution to provide only a single message accompanying the outcome. That one can be pushed to the `string[] outcomeMsg` of the rule. 

So, I changed the type within the rule type definition. Want to give it a check for a 👍 ?